src/raft_log.rs

```diff
@@ -354,12 +373,12 @@ impl<T: Storage> RaftLog<T> {
         term > self.last_term() || (term == self.last_term() && last_index >= self.last_index())
     }
 
-    /// Returns any entries since the a particular index.
+    /// Returns committed and persisted entries since max(`since_idx` + 1, first_index).
     pub fn next_entries_since(&self, since_idx: u64) -> Option<Vec<Entry>> {
         let offset = cmp::max(since_idx + 1, self.first_index());
-        let committed = self.committed;
-        if committed + 1 > offset {
-            match self.slice(offset, committed + 1, None) {
+        let high = cmp::min(self.committed, self.persisted) + 1;
```

Comment:
Case `self.committed > self.persisted` is not covered in this file.

Reply:
I merge the `test_has_next_ents` and `test_next_ents` and add more cases